src/fsharp/TastOps.fs

```diff
@@ -2707,12 +2718,36 @@ let fullDisplayTextOfValRefAsLayout   (vref:ValRef) =
         pathText ^^ SepL.dot ^^ wordL n
         //pathText +.+ vref.DisplayName
 
-
 let fullMangledPathToTyconRef (tcref:TyconRef) = 
     match tcref with 
     | ERefLocal _ -> (match tcref.PublicPath with None -> [| |] | Some pp -> pp.EnclosingPath)
     | ERefNonLocal nlr -> nlr.EnclosingMangledPath
-  
+    
+/// generates a name like 'System.IComparable<System.Int32>.Get'
+/// for types in the global namespace, `global is prepended (note the backtick)
+let qualifiedInterfaceImplementationName g (tt:TType) memberName =
+    let rec TType_ToCodeLikeString (x:TType) : string =
+        let TyconRefToFullName (tc:TyconRef) =
+            let namespaceParts =
+                // we need to ensure there are no collisions between (for example)
+                // - ``IB<GlobalType>`` (non-generic)
+                // - IB<'T> instantiated with 'T = GlobalType
+                // This is only an issue for types inside the global namespace, because '.' is invalid even in a quoted identifier.
+                // So if the type is in the global namespace, prepend 'global`', because '`' is also illegal -> there can be no quoted identifer with that name.
+                match fullMangledPathToTyconRef tc with
+                | [||] -> [| "global`" |]
+                | ns -> ns
+            seq { yield! namespaceParts; yield tc.DisplayName } |> String.concat "."
+        match x with
+        | TType_app (a,[]) -> TyconRefToFullName a
+        | TType_app (a,b) ->
+            let genericParameters = b |> Seq.map (stripTyEqnsAndErase true g >> stripMeasuresFromTType g >> TType_ToCodeLikeString) |> String.concat ", "
+            sprintf "%s<%s>" (TyconRefToFullName a) genericParameters
+        | TType_var (v) -> "'" + v.Name
+        | _ -> failwithf "unexpected: expected TType_app but got %O" (x.GetType())
```

Comment:
This line is where the other types will blow up. My hypothesis is, if you have `interface ITest<int * int>`, it will hit this `failwithf "unexpected ...."`.

Reply:
yeah, that'll probably break.

Reply:
I looked at it again, and I think it should actually work as is: ``stripTyEqnsAndErase`` has this comment:

```f#
/// This erases outermost occurrences of inference equations, type abbreviations, non-generated provided types
/// and measureable types (float<_>).
/// It also optionally erases all "compilation representations", i.e. function and
/// tuple types, and also "nativeptr<'T> --> System.IntPtr"
```

It doesn't mention anonymous records, so let's see what CI says to the unit test.

Reply:
Everything except anons seems to work (see most recent commit and xi), I'll look at it again tomorrow, but I'd propose to add a flag to "stripTyEqnsAndErase" that also erases anons.